mod-infra/src/main/scala/at/forsyte/apalache/infra/passes/SourceOption.scala

```diff
@@ -0,0 +1,14 @@
+package at.forsyte.apalache.infra.passes
+
+/** Defines the data sources supported in the [[PassOptions]] */
+object SourceOption {
+
+  /** Supported data sources */
+  sealed trait T
+
+  /** Data to be loaded from a file */
+  case class File(file: java.io.File) extends T
+
+  /** Data supplied as a string */
+  case class String(content: java.lang.String) extends T
+}
```

Comment:
This isn't an idiomatic way to define these kinds of ADTs in Scala, afaict, but searching google and looking through  our code base, I couldn't find any obvious, principled convention for how to name case classes to implement a concise and readable sum type.

Pointers on how this should look are welcome.